packages/ckeditor5-engine/tests/model/utils/deletecontent.js

```diff
@@ -283,8 +322,20 @@ describe( 'DataController utils', () => {
 
 			test(
 				'leaves just one element when all selected',
+				'<paragraph>[x</paragraph><paragraph>foo</paragraph><paragraph>y]bar</paragraph>',
+				'<paragraph>[]bar</paragraph>'
+			);
+
+			test(
+				'leaves just one element when all selected (different name)',
```

Comment:
I think that this test should reflect what merging happen.

Reading the test description I'm not sure what is the expected outcome and how this test differs from `leaves just one (first) element when all selected (different name)`.

Why first not last?